wavetorch/core/cell.py

```diff
@@ -121,7 +123,8 @@ def __init__(self, dt : float, Nx : int, Ny : int,
         if design_region is not None:
             # Use specified design region
             assert design_region.shape == (Nx, Ny), "Design region mask dims must match spatial dims"
-            self.register_buffer("design_region", design_region * (self.b_boundary == 0))
+            # self.register_buffer("design_region", design_region * (self.b_boundary == 0))
+            self.register_buffer("design_region", design_region * (self.b_boundary == 0).type(torch.uint8))
```

Comment:
This is probably best practice for a mask, but I just want to make sure... were you actually encountering an error with the old code?

Reply:
I was. A type error.